src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Symbolic/DerivativeEffect.cs

```diff
@@ -0,0 +1,26 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.RegularExpressions.Symbolic
+{
+    /// <summary>
+    /// Describes effects to be applied to registers.
+    /// </summary>
+    internal struct DerivativeEffect
+    {
+        public enum EffectKind
+        {
+            CaptureStart,
+            CaptureEnd,
+        };
+
+        public EffectKind Kind;
+        public int IntArg0;
+
+        public DerivativeEffect(EffectKind kind, int intArg0)
```

Comment:
No better name than "int arg 0"? :smile:  Is the issue that its meaning changes based on the kind of the effect?  Maybe an XML comment for the parameter that says something like "When kind is xyz, this arguments means blah. When kind is abc, this argument means bleh." Etc.

Reply:
The very general naming is perhaps too forward looking right now. For both the effects kinds it is currently the capture number. I'll make it more specific.